packages/design-system-react/src/components/button-icon/ButtonIcon.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+import { twMerge } from '../../utils/tw-merge';
+import { ButtonBase } from '../button-base';
+import { Icon } from '../icon';
+import {
+  BUTTON_ICON_SIZE_CLASS_MAP,
+  BUTTON_ICON_SIZE_TO_ICON_SIZE_CLASS_MAP,
+} from './ButtonIcon.constants';
+import type { ButtonIconProps } from './ButtonIcon.types';
+import { ButtonIconSize } from './ButtonIcon.types';
+
+export const ButtonIcon = React.forwardRef<HTMLButtonElement, ButtonIconProps>(
+  (
+    {
+      className,
+      iconName,
+      iconProps,
+      ariaLabel,
+      isDisabled = false,
+      isInverse = false,
+      isFloating = false,
+      size = ButtonIconSize.Md,
+      style,
+      ...props
+    },
+    ref,
+  ) => {
+    const isInteractive = !isDisabled;
+
+    const mergedClassName = twMerge(
+      // Base styles
+      'p-0',
+      // Size styles
+      BUTTON_ICON_SIZE_CLASS_MAP[size],
+      // Floating styles
+      isFloating && [
+        'rounded-full',
+        !isInverse && 'bg-icon-default text-background-default',
+        isInverse && 'text-icon-default bg-background-default',
```

Comment:
Default `isInverse` + `isFloating` to the isFloating default
```suggestion
        isInverse && 'text-icon-default bg-background-default',
```